modules/renter/contractor/renew.go

```diff
@@ -76,6 +84,13 @@ func (c *Contractor) managedRenewContracts() error {
 
 	c.log.Printf("renewing %v contracts", len(renewSet))
 
+	// If this is the first time we are renewing contracts in a new allowance
+	// period, reset the spending metrics.
+	// TODO: how do we determine this?
+	if true {
```

Comment:
Ideally you don't have to figure this out, whatever function is figuring out how much is unspent will just add up all the costs of all the contracts in the current period. Which would mean that from a contract you need to be able to determine:

- how much it cost
- what period it belongs in (based on the block it was created in)

----------

If that approach ends up being too big of a change, you'd know you are in a new period by having some way to check what period you are currently in. If (currentPeriod.StartHeight+periodLength < currentHeight) then you are moving to a new period. You need to reset the tracking and set the currentPeriod.StartHeight = currentHeight.